Dart/src/com/jetbrains/lang/dart/psi/impl/DartUriElementBase.java

```diff
@@ -53,6 +57,25 @@ public PsiReference getReference() {
     final String uri = ((DartUriBasedDirective)parent).getUriString();
     final int uriOffset = ((DartUriBasedDirective)parent).getUriStringOffset();
 
+    if (DartResolver.isServerDrivenResolution()) {
+      final PsiFile refPsiFile = getContainingFile();
+      final int refOffset = getTextOffset();
+      final int refLength = getTextLength();
+      final PluginNavigationRegion region = DartResolver.INSTANCE.findRegion(refPsiFile, refOffset, refLength);
+      if (region != null) {
+        final PluginNavigationTarget target = region.getTargets().get(0);
+        final String targetPath = target.getFile();
+        final VirtualFile targetVirtualFile = LocalFileSystem.getInstance().findFileByPath(targetPath);
```

Comment:
I'm sorry, but I don't understand.
What should I return if it cannot be resolved?

There were another branches in this method that `return PsiReference.EMPTY_ARRAY` when they cannot resolve an URI.